kiwi/system/result.py

```diff
@@ -15,45 +15,53 @@
 # You should have received a copy of the GNU General Public License
 # along with kiwi.  If not, see <http://www.gnu.org/licenses/>
 #
-from collections import namedtuple
 import logging
 import pickle
 import os
+from typing import (
+    Dict, NamedTuple, TypeVar
+)
 
 # project
+from kiwi.xml_state import XMLState
 from kiwi.exceptions import (
     KiwiResultError
 )
 
 log = logging.getLogger('kiwi')
 
 # must be global to allow pickle to find it
-result_file_type = namedtuple(
-    'result_file_type', ['filename', 'use_for_bundle', 'compress', 'shasum']
+ResultFileType = NamedTuple(
+    'ResultFileType', [('filename', str),
+                       ('use_for_bundle', bool),
+                       ('compress', bool),
+                       ('shasum', bool)]
 )
+ResultType = TypeVar('ResultType', bound='Result')
```

Comment:
I did not understand why we need a new global ResultType variable from the Result class. Is a self reference to the Result class not possible when used as return type in ```def load(filename: str) -> ResultType:``` ?

Reply:
meaning ```def load(filename: str) -> Result:``` does not work ?